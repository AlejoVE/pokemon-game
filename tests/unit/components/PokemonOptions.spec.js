import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'
import { pokemons } from '../mocks/pokemons'
describe('PokemonOptions Component', () => {
	let wrapper

	beforeEach(() => {
		wrapper = shallowMount(PokemonOptions, {
			props: {
				pokemons,
				selected: false,
			},
		})
	})

	test('should match snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	test('should show 4 options correctly', () => {
		const liTags = wrapper.findAll('li')
		expect(liTags.length).toBe(4)

		expect(liTags[0].text()).toBe('bulbasaur')
		expect(liTags[1].text()).toBe('ivysaur')
		expect(liTags[2].text()).toBe('venusaur')
		expect(liTags[3].text()).toBe('charmander')
	})

	test('should emit event "selection"', () => {
		const [li1] = wrapper.findAll('li')
		li1.trigger('click')

		expect(wrapper.emitted('selection').length).toBe(1)
		expect(wrapper.emitted('selection')[0]).toEqual([1])
	})
})
