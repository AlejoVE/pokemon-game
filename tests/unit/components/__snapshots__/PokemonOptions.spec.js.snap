// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should match snapshot 1`] = `
<div class="options-container">
  <ul>
    <li class="">bulbasaur</li>
    <li class="">ivysaur</li>
    <li class="">venusaur</li>
    <li class="">charmander</li>
  </ul>
</div>
`;
